cmd/vl3-nse/service_registry.go

```diff
@@ -23,9 +25,13 @@ const (
 
 type validationErrors []error
 
-func NewServiceRegistry(addr string) (ServiceRegistry, ServiceRegistryClient, error) {
+func NewServiceRegistry(addr string, ctx context.Context) (ServiceRegistry, ServiceRegistryClient, error) {
+	tlscfg, err := tools.GetConfig().SecurityProvider.GetTLSConfig(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
 	conn, err := grpc.Dial(addr,
-		grpc.WithInsecure(),
+		grpc.WithTransportCredentials(credentials.NewTLS(tlscfg)),
```

Comment:
A check has to be added for the env var ``INSECURE`` and according to the value of this, we should either enable a secure GRPC connection or not.